File: task-03/SubTask3/diamond_pattern.exs
n = IO.gets("Enter a number: ") |> String.trim() |> String.to_integer()
for i <- 0..(n - 1) do
  IO.puts String.duplicate(" ", n - i - 1) <> String.duplicate("*", 2 * i + 1)
end

for i <- (n - 2)..0 do
  IO.puts String.duplicate(" ", n - i - 1) <> String.duplicate("*", 2 * i + 1)
end

File: task-03/SubTask3/diamond_pattern.rs
use std::io;

fn main() {
    let mut input = String::new();
    println!("Enter a number: ");
    io::stdin().read_line(&mut input).expect("Failed to read line");
    let n: usize = input.trim().parse().expect("Please type a number!");

    for i in 0..n
    {
        println!("{:width$}{}", "", "*".repeat(2 * i + 1), width = n - i - 1);
    }
    for i in (0..n-1).rev() 
    {
        println!("{:width$}{}", "", "*".repeat(2 * i + 1), width = n - i - 1);
    }
}
